src/test/versionbits_tests.cpp

```diff
@@ -44,6 +44,12 @@ class TestConditionChecker : public AbstractThresholdConditionChecker
     int GetStateSinceHeightFor(const CBlockIndex* pindexPrev) const { return AbstractThresholdConditionChecker::GetStateSinceHeightFor(pindexPrev, paramsDummy, cache); }
 };
 
+class TestDelayedActivationConditionChecker : public TestConditionChecker
+{
+public:
+    int MinActivationHeight(const Consensus::Params& params) const override { return 15000; }
```

Comment:
Could be done, personally indifferent -- the main difference between why the others are explicit in `TestConditionChecker` and `MinActivationHeight` is not is that `MinActivationHeight` is defined to have a sane default function, whereas the others are not.

Reply:
I think it would be better for the `ConditionChecker` to not be abstract and just decode the `Consensus::Params` structure directly, so that tests just pass in different data, but are running the same chunks of code. Leaving this as-is until later refactoring either way though.